README.md

```diff
@@ -7,7 +7,261 @@
 A Visual Studio Code extension for [ty](https://github.com/astral-sh/ty), an extremely fast
 Python type checker and language server, written in Rust.
 
-**This project is still in development and is not ready for production use.**
+> [!WARNING]
+>
+> This extension uses the preview version of ty and is not ready for production use.
+>
+> We're working hard to make ty stable and feature-complete, but until then, expect to encounter bugs,
+> missing features, and fatal errors.
+
+The extension ships with `ty==0.0.0a7`.
+
+## Features
+
+Currently, the extension supports the following features:
+
+- Showing diagnostics in the editor
+- Hover type information for symbols
+- Go to type definition
+- Inlay hints for assignment targets
+- [Completions](#completionsenable)
+
+> [!NOTE]
+>
+> If you want to test the language server features like completions, go to type definition, etc., it's
+> recommended to disable the language server from the [Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python) (if installed) by adding the
+> following [setting](https://code.visualstudio.com/docs/python/settings-reference#_intellisense-engine-settings) to your `settings.json`:
+>
+> ```json
+> {
+>   "python.languageServer": "None"
+> }
+> ```
+
+## Usage
+
+Once installed in Visual Studio Code, ty will automatically execute when you open or edit a
+Python or Jupyter Notebook file.
+
+If you want to disable ty, you can [disable this extension](https://code.visualstudio.com/docs/editor/extension-marketplace#_disable-an-extension)
+per workspace in Visual Studio Code.
+
+## Untrusted Workspace
+
+The extension supports loading in an [untrusted workspace](https://code.visualstudio.com/docs/editor/workspace-trust).
+When the workspace is untrusted, the extension will always use the bundled executable of
+the `ty` binary regardless of any other settings.
+
+The following settings are not supported in an untrusted workspace:
+
+- [`ty.importStrategy`](#importstrategy)
+- [`ty.interpreter`](#interpreter)
+- [`ty.path`](#path)
+
+## Settings
```

Comment:
An alternative to enumerating the settings here would be to include them in the ty repository: https://github.com/astral-sh/ty/pull/343

My preference would be to include it in ty repository instead of here but happy to hear others thoughts.

Reply:
Documenting the LSP specific settings in the ty repository makes sense to me. I think I'd prefer having the VS code specific settings documented here. Or does that require more repetition because the VS code extension has many settings in common with the LSP?

Reply:
Yeah, I think that split could work as well. I mainly did it this way to keep it consistent with how it is with Ruff.